pkg/providers/instance/azure_client.go

```diff
@@ -112,7 +112,7 @@ func NewAZClient(ctx context.Context, cfg *auth.Config, env *azure.Environment)
 	if err != nil {
 		return nil, err
 	}
-
+	cred = auth.NewTokenWrapper(cred)
```

Comment:
I suppose these are _generally_ temporary until things are fixed on the other side. Could you add some comments about that?

Reply:
Ideally we could remove this code entirely so I have it separated. Also we need it to be generic so any team can pass in a token without the same azure config logic we have going on. 

The plan is for some other teams to yoink the code so coupling with azureconfig is not ideal. Instead we should integrate the wrapper with just the token so that anyone with an auth token can use the code.